src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/BurdenEstimateWriter.kt

```diff
@@ -70,37 +71,38 @@ open class BurdenEstimateWriter(private val readDatabaseDSL: DSLContext,
         val outcomeLookup = getOutcomesAsLookup()
         val modelRuns = getModelRunsAsLookup(setId)
         val modelRunParameterId = getModelRunParameterSetId(setId)
-        val writeDatabaseDSL = writeDatabaseDSLPromise()
-
-        // The only foreign keys are:
-        // * burden_estimate_set, which is the same for every row, and it's the one we just created and know exists
-        // * country, which we check below, per row of the CSV (and each row represents multiple rows in the database
-        //   so this is an effort saving).
-        // * burden_outcome, which we check below (currently we check for every row, but given these are set in the
-        //   columns and don't vary by row this could be made more efficient)
-        writeDatabaseDSL.withoutCheckingForeignKeyConstraints(table) {
-
-            PipedOutputStream().use { stream ->
-                // First, let's set up a thread to read from the stream and send
-                // it to the database. This will block if the thread is empty, and keep
-                // going until it sees the Postgres EOF marker.
-                val inputStream = PipedInputStream(stream).buffered()
-                val writeToDatabaseThread = writeStreamToDatabase(inputStream, writeDatabaseDSL)
-
-                // In the main thread, write to piped stream, blocking if we get too far ahead of
-                // the other thread ("too far ahead" meaning the buffer on the input stream is full)
-                writeCopyData(
-                        outcomeLookup,
-                        countries,
-                        modelRuns,
-                        modelRunParameterId,
-                        stream,
-                        estimates,
-                        expectedDisease,
-                        setId)
-
-                // Wait for the worker thread to finished
-                writeToDatabaseThread.join()
+        writeDatabaseDSLPromise.inside { writeDatabaseDSL ->
```

Comment:
Only change here is indentation due to wrapping everything in this new call to "inside".